src/main/java/formflow/library/ScreenController.java

```diff
@@ -276,19 +281,8 @@ ModelAndView updateOrCreateIteration(
     FormSubmission formSubmission = new FormSubmission(formData);
     String subflowName = currentScreen.getSubflow();
     Submission submission = findOrCreateSubmission(httpSession, flow);
-
     actionManager.handleOnPostAction(currentScreen, formSubmission, submission, iterationUuid);
 
-    if (isNewIteration) {
-      // handle start iteration page, if new flow
-      HashMap<String, SubflowConfiguration> subflows = getFlowConfigurationByName(flow).getSubflows();
-      subflowName = subflows.entrySet().stream()
-          .filter(subflow -> subflow.getValue().getIterationStartScreen().equals(screen))
-          .map(Entry::getKey)
-          .findFirst()
-          .orElse(null);
-    }
```

Comment:
While testing we unintentionally added two subflows with the start screens of the same name, which resulted in the `findFirst` grabbing the wrong one. After some investigation, we determined that since the subflowName is being set with `currentScreen.getSubflow()`, that value should always be present and this block is not needed.